external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -91,11 +91,31 @@ private[avro] case class AvroDataToCatalyst(
         null
     }
 
+  /**
+   * This computes the offset for avro binary decoding based on the presence of a magic number
+   * defined in avro BinaryMessageEncoder.java as of now we are assuming that the binary
+   * provided was encoded via avro BinaryEncoder.java. However some serializers use
```

Comment:
I wonder how safe it is to assume that if the two bytes match the magic number, they're the magic number? would a regular Avro file without it ever start with these bytes?